examples/wip/basis/app.js

```diff
@@ -1,78 +1,108 @@
 /* global document, URL */
 import {load} from '@loaders.gl/core';
-import {BasisLoader} from '@loaders.gl/basis';
-
-const btn = document.createElement('button');
-document.body.appendChild(btn);
-btn.innerText = 'Load Basis File';
+import {BasisLoader, BasisFormat} from '@loaders.gl/basis';
 
 const canvas = document.createElement('canvas');
 document.body.appendChild(canvas);
+canvas.width = 768;
+canvas.height = 512;
+canvas.style.display = 'block';
 const gl = canvas.getContext('webgl');
 const dxtSupported = !!gl.getExtension('WEBGL_compressed_texture_s3tc');
+const loadOptions = {format: BasisFormat.cTFRGB565};
+
+if (dxtSupported) {
+  loadOptions.format = {
+    alpha: BasisFormat.cTFBC3,
+    noAlpha: BasisFormat.cTFBC1
+  };
+}
+
+const vs = `
+precision highp float;
+
+attribute vec2 position;
+varying vec2 uv;
+
+void main() {
+  gl_Position = vec4(position, 0.0, 1.0);
+  uv = vec2(position.x * .5, -position.y * .5) + vec2(.5, .5);
+}
+`;
+
+const fs = `
+precision highp float;
+
+uniform sampler2D tex;
+varying vec2 uv;
+
+void main() {
+  gl_FragColor = vec4(texture2D(tex, uv).rgb, 1.);
+}
+`;
+
+const vertexShader = gl.createShader(gl.VERTEX_SHADER);
+gl.shaderSource(vertexShader, vs);
+gl.compileShader(vertexShader);
+
+const fragmentShader = gl.createShader(gl.FRAGMENT_SHADER);
+gl.shaderSource(fragmentShader, fs);
+gl.compileShader(fragmentShader);
 
-const BASIS_FORMAT = {
-  cTFETC1: 0,
-  cTFETC2: 1,
-  cTFBC1: 2,
-  cTFBC3: 3,
-  cTFBC4: 4,
-  cTFBC5: 5,
-  cTFBC7_M6_OPAQUE_ONLY: 6,
-  cTFBC7_M5: 7,
-  cTFPVRTC1_4_RGB: 8,
-  cTFPVRTC1_4_RGBA: 9,
-  cTFASTC_4x4: 10,
-  cTFATC_RGB: 11,
-  cTFATC_RGBA_INTERPOLATED_ALPHA: 12,
-  cTFRGBA32: 13,
-  cTFRGB565: 14,
-  cTFBGR565: 15,
-  cTFRGBA4444: 16,
-};
+const program = gl.createProgram();
+gl.attachShader(program, vertexShader);
+gl.attachShader(program, fragmentShader);
+gl.linkProgram(program);
+
+if (!gl.getProgramParameter(program, gl.LINK_STATUS)) {
+  console.error(gl.getProgramInfoLog(program));
+} else {
+  gl.enableVertexAttribArray(0);
+  const buffer = gl.createBuffer();
+  gl.bindBuffer(gl.ARRAY_BUFFER, buffer);
+  gl.bufferData(gl.ARRAY_BUFFER, new Float32Array([-1, -1, -1, 1, 1, -1, 1, 1]), gl.STATIC_DRAW);
+  gl.vertexAttribPointer(0, 2, gl.FLOAT, false, 0, 0);
+
+  gl.useProgram(program);
+}
+
+function renderTexture({width, height, format, decodedData}) {
+  if (!dxtSupported) {
+    // TODO: implement cTFRGB565 support
+    return;
+  }
+
+  const ext = gl.getExtension('WEBGL_compressed_texture_s3tc');
+  const glFormat = {
+    [BasisFormat.cTFBC1]: ext.COMPRESSED_RGB_S3TC_DXT1_EXT,
+    [BasisFormat.cTFBC3]: ext.COMPRESSED_RGBA_S3TC_DXT5_EXT
+  }[format];
+
+  const texture = gl.createTexture();
+  gl.bindTexture(gl.TEXTURE_2D, texture);
+  gl.compressedTexImage2D(gl.TEXTURE_2D, 0, glFormat, width, height, 0, decodedData);
```

Comment:
This will fail if there's more than 1 mip level (decodedData will be the wrong size).

Reply:
It works: every level will get its own `compressedTexImage2D()` call with separate data.